src/sage/combinat/designs/difference_family.py

```diff
@@ -1750,29 +1826,30 @@ def supplementary_difference_set_from_rel_diff_set(q, existence=False, check=Tru
     for d in set1:
         hall += P.monomial(d[0])
 
-    T_2m = 0
-    for i in range(2 * m):
-        T_2m += P.monomial(i)
+    def get_T(k):
+        T = P.monomial(0) -1
+        for i in range(k):
+            T += P.monomial(i)
+        return T
```

Comment:
Why this change? Having this as a function instead of just being computed once seems slower to me.

Reply:
This function is also used later (line 1828) with a different value of ``k``.